client/client.go

```diff
@@ -163,44 +170,231 @@ func makeURL(base string, parts []string) string {
 	return base + strings.Join(parts, "/")
 }
 
-func (c *client) MakeServiceURL(serviceType string, parts []string) (string, error) {
+func (c *client) MakeServiceURL(serviceType, apiVersion string, parts []string) (string, error) {
 	return makeURL(c.baseURL, parts), nil
```

Comment:
In theory, I agree.  In practice, I’m not sure there are OpenStack deployments without username/password or token required?  None of the current goose, nor juju openstack provider test code uses a non authenticating client.  Only the goose client tests do in cases to do with containers, Swift doesn’t use versions.

I need some input about what is supported with openstack deployments used by juju.